File: byteback-test/test-scala/src/main/scala/byteback/test/exceptions/Basic.scala
/**
 * RUN: %{byteback} -m -cp %{jar} -c %{class} -o %t.bpl
 */
package byteback.test.exceptions;

import byteback.annotations.Contract._;
import byteback.annotations.Special._;
import byteback.annotations.Operator._;
import byteback.annotations.Operator.{eq => equal};

class Basic {

  class ExceptionA extends Exception

  class ExceptionB extends Exception

  def tryCatchBlock(): Unit = {
    try {
      throw new Exception();
    } catch {
      case _: Exception =>
    }
  }

  @Return
  def neverThrows(): Unit = {
  }

  def neverCatches(): Unit = {
    try {
      neverThrows();
    } catch {
      case _: Exception => assertion(false);
    }
  }

  @Predicate
  def always_throws(e: Throwable): Boolean = {
    return e.isInstanceOf[Exception];
  }

  @Ensure("always_throws")
  def alwaysThrows(): Unit = {
    throw new Exception();
  }

  def alwaysCatches(): Unit = {
    try {
      alwaysThrows();
      assertion(false);
    } catch {
      case _: Throwable =>
    }
  }

}
/**
 * RUN: %{verify} %t.bpl | filecheck %s
 * CHECK: Boogie program verifier finished with 6 verified, 0 errors
 */

File: byteback-test/test-java-17/src/main/java/byteback/test/switchexpression/Basic.java
/**
 * RUN: %{byteback} -cp %{jar} -c byteback.test.switchexpression.Basic -o %t.mlw
 */
package byteback.test.switchexpression;

import static byteback.annotations.Contract.*;
import static byteback.annotations.Operator.*;

public class Basic {

	public static int state = 1;

	public static void switchInteger() {
		state = 1;
		int a = switch(state) { case 1 -> 1; case 2 -> 2; default -> 3; };

		assertion(eq(a, 1));
	}

	public static void switchYieldInteger() {
		state = 1;
		int a = switch(state) { case 1 -> { yield(1); } case 2 -> { yield(2); } default -> { yield(3); } };

		assertion(eq(a, 1));
	}

}
/**
 * RUN: %{verify} %t.mlw
 * CHECK-IGNORE: Boogie program verifier finished with 4 verified, 0 errors
 */
